types/decimal.go

```diff
@@ -209,11 +211,12 @@ func (d Dec) ToLeftPadded(totalDigits int8) string {
 //   BankerRoundChop(1500, 3) = 2
 func BankerRoundChop(d *big.Int, n int64) (chopped *big.Int) {
 
+	negated := (d.Sign() == -1)
 	// remove the negative and add it back when returning
-	if d.Sign() == -1 {
+	if negated {
 		d = new(big.Int).Neg(d)
```

Comment:
This comment got hidden I think,
Chris asked if we can do this in place, and I asked if we could do this without the defer.

Reply:
clever iterative solution, thanks.